### PR TITLE
Bug #516 - Use latest Tycho and Java 17 for building

### PR DESCRIPTION
**Pull request by lcaron**
Move Tycho version (and move to JDK 17)

### CODE REVIEW

File: .mvn/extensions.xml [hunk 1]
```diff
@@ -2,6 +2,6 @@
   <extension>
     <groupId>org.eclipse.tycho.extras</groupId>
     <artifactId>tycho-pomless</artifactId>
-    <version>2.5.0</version>
+    <version>3.0.4</version>
```
Comment (laeubi) [inline .mvn/extensions.xml:5]:
```suggestion
    <version>4.0.4</version>
```
The latest Tycho version as of today is [4.0.4](https://github.com/eclipse-tycho/tycho/releases/tag/tycho-4.0.4)

File: releng/org.eclipse.nebula.nebula-parent/pom.xml [hunk 2]
```diff
@@ -22,7 +22,7 @@ Contributors:
 
 	<properties>
 
-		<tycho-version>2.6.0</tycho-version>
+		<tycho-version>3.0.4</tycho-version>
```
Comment (laeubi) [inline releng/org.eclipse.nebula.nebula-parent/pom.xml:25]: You might want to use a `maven.config` file and then share the Tycho version for extension/pom in that file like described here: https://tycho.eclipseprojects.io/doc/latest/StructuredBuild.html#pomless-builds

**Comment by laeubi** [comment]
There is no `<tycho-extras-version>` anymore for a long time so you probably want to replace that with search+replace

**Comment by Boarschti42** [comment]
Hi, i recommend using 
```
<extensions>
    <extension>
        <groupId>org.eclipse.tycho</groupId>
        <artifactId>tycho-build</artifactId>
        <version>4.0.4</version>
    </extension>
</extensions>
```

instead of tycho-pomless since this is the current way suggested.

**Comment by merks** [comment]
This is obsolete with the changes from https://github.com/eclipse/nebula/issues/559

**Pull request closed without merging**